aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -249,4 +246,42 @@ static TensorIterator make_reduction(
   return make_reduction(name, result1, result2, self, dim, keepdim, dtype, dtype);
 }
 
+static void zero_numel_check_dims(const Tensor& self, int64_t dim) {
+  if (self.ndimension() == 0) {
+    TORCH_CHECK_INDEX(dim == 0 || dim == -1, "Expected reduction dim -1 or 0 for scalar but got ", dim);
```

Comment:
@v0dro what I meant here is that this function should be:
```suggestion
+ static void zero_numel_check_dims(const Tensor& self, int64_t dim, const char* const fn_name) {
+  if (self.ndimension() == 0) {
+    TORCH_CHECK_INDEX(dim == 0 || dim == -1, fn_name, ": Expected reduction dim -1 or 0 for scalar but got ", dim);
```
and same in the other `TORCH_CHECK_INDEX`. Sorry for the misunderstanding.

These errors are user-facing errors, so they should be as informative as possible. As such, the names reported in these errors should be those of the functions that the user calls whenever possible (see the `_minmax_all()` function.

Reply:
Ah I see. I misunderstood your previous comment. Will fix it now.

Reply:
So @Lezcano I have an argument against having to include the user-facing function in the error.
1. When this error is raised the user can simply check the Python stack trace to see which function has failed. The error provides enough data about malformed inputs.
2. In case a developer encounters this error, they probably want to quickly search for the specific name of the function within the C code that raised that error and not have to trace the calling function  to this function.
3. If a developer wants to know the caller, they can just set a breakpoint over this `TORCH_CHECK_INDEX` and run it inside GDB.
4. Therefore, I think the current arrangement in which is the name of the function that actually raises the error is displayed is appropriate for most use cases.

Reply:
@mruberry was the one pushing for these changes. I am completely in favour of giving the name of the calling function when throwing an exception, as I find it very useful as well in terms of usability. I do not think that "checking the trace" is the best of ways to approach this.

Reply:
Following up on what @Lezcano said, we think it's a better user experience if the function they call is what throws the error. Otherwise we're revealing implementation details to the user, which can often be confusing as they seem unrelated to their action.

Reply:
Alright I've updated the functions with the suggested changes.